File: The Torture Chamber.java
import java.io.*;

class Reader {

    final private int BUFFER_SIZE = 1 << 16;
    private DataInputStream din;
    private byte[] buffer;
    private int bufferPointer, bytesRead;
    private int http;

    public Reader() {
        din = new DataInputStream(System.in);
        buffer = new byte[BUFFER_SIZE];
        bufferPointer = bytesRead = 0;
    }

    public Reader(String file_name) throws IOException {
        din = new DataInputStream(new FileInputStream(file_name));
        buffer = new byte[BUFFER_SIZE];
        bufferPointer = bytesRead = 0;
    }

    public String readLine() throws IOException {
        byte[] buf = new byte[64]; // line length
        int cnt = 0, c;
        while ((c = read()) != -1) {
            if (c == '\n') {
                break;
            }
            buf[cnt++] = (byte) c;
        }
        return new String(buf, 0, cnt);
    }

    public int nextInt() throws IOException {
        int ret = 0;
        byte c = read();
        while (c <= ' ') {
            c = read();
        }
        boolean neg = (c == '-');
        if (neg) {
            c = read();
        }
        do {
            ret = ret * 10 + c - '0';
        } while ((c = read()) >= '0' && c <= '9');

        if (neg) {
            return -ret;
        }
        return ret;
    }

    public long nextLong() throws IOException {
        long ret = 0;
        byte c = read();
        while (c <= ' ') {
            c = read();
        }
        boolean neg = (c == '-');
        if (neg) {
            c = read();
        }
        do {
            ret = ret * 10 + c - '0';
        } while ((c = read()) >= '0' && c <= '9');
        if (neg) {
            return -ret;
        }
        return ret;
    }

    public double nextDouble() throws IOException {
        double ret = 0, div = 1;
        byte c = read();
        while (c <= ' ') {
            c = read();
        }
        boolean neg = (c == '-');
        if (neg) {
            c = read();
        }

        do {
            ret = ret * 10 + c - '0';
        } while ((c = read()) >= '0' && c <= '9');

        if (c == '.') {
            while ((c = read()) >= '0' && c <= '9') {
                ret += (c - '0') / (div *= 10);
            }
        }

        if (neg) {
            return -ret;
        }
        return ret;
    }

    private void fillBuffer() throws IOException {

        bytesRead = din.read(buffer, bufferPointer = 0, BUFFER_SIZE);
        if (bytesRead == -1) {
            buffer[0] = -1;
        }
    }

    private byte read() throws IOException {
        if (bufferPointer == bytesRead) {
            fillBuffer();
        }
        return buffer[bufferPointer++];
    }

    public void close() throws IOException {
        if (din == null) {
            return;
        }
        din.close();
    }
}

public class TheTortureChamber {
	public static void main(String[] args) throws IOException {
	    Reader r = new Reader();
	    
		long a = r.nextLong();
		long b = r.nextLong();

		int diff1 = (int) (b-a);
		int diff2 = (int) Math.sqrt(b)+1;

		boolean[] prime1 = new boolean[diff1];
		boolean[] prime2 = new boolean[diff2];

		for(int i=0; i < diff1; i++)
			if(i+a > 3 && ((i+a)%2 == 0 || (i+a)%3 == 0))
				prime1[i] = true;

		for(int i=5; i<diff2; i+=6)
			for(int j=i; j<=i+2; j+=2)
				if(j < diff2 && !prime2[j]) {
					for(int k=j*j; 0<=k && k<diff2; k+=j)
						prime2[k] = true;
	
					for(int l = (int) (a/j*j-a); l<diff1; l+=j)
						if(l+a != j && l >= 0)
							prime1[l] = true;
				}

		int ans = 0;
		for(int i=0; i<diff1; i++)
			if(!prime1[i])
				ans++;

		System.out.println(ans);
               
	}
}

File: DMOJ/Fun Math/The Torture Chamber.cpp
#include<iostream>
#include <stdio.h>
#include <math.h> 
#define ULL unsigned long long
#define LL long long
#define scan(x) do{while((x=getchar_unlocked())<'0'); for(x-='0'; '0'<=(_=getchar_unlocked()); x=(x<<3)+(x<<1)+_-'0');}while(0)
char _;

using namespace std;

ULL a, b;
LL diff1, diff2;
bool prime1 [1000000];
bool prime2 [20000100];

int main () {
scan(a);
scan(b);
diff1 = b-a;
diff2 = sqrt(b)+1;

for (int i=0; i<diff1; ++i)
  if (i + a > 3 && ((i+a) % 2 == 0 || (i + a) % 3 == 0))
    prime1[i] = true;
    
    for (int i=5; i<diff2; i += 6)
      for (int j=i; j<= i+2; j += 2)
        if (j < diff2 && !prime2[j]){
          for (int k= j * j; 0<=k && k<diff2; k += j)
            prime2[k] = true;
            
            for (int l= (int) (a / j*j - a); l<diff1; l += j)
              if (l + a != j && l >= 0)
                prime1[l] = true;
        }
        LL cnt = 0;
        for (int i=0; i<diff1; i++)
          if (!prime1[i])
            cnt++;
            
        cout << cnt << "\n";
}